dashboard/src/components/AppList/AppList.test.tsx

```diff
@@ -219,20 +220,12 @@ it("clicking 'List All' checkbox should trigger toggleListAll", () => {
     ],
     listingAll: false,
   } as IAppState;
-  const wrapper = shallow(
-    <AppList
-      {...defaultProps}
-      apps={apps}
-      toggleListAll={jest.fn((toggle: boolean) => {
-        apps.listingAll = toggle;
-      })}
-    />,
-  );
+  const wrapper = shallow(<AppList {...defaultProps} apps={apps} />);
   const checkbox = wrapper.find('input[type="checkbox"]');
   expect(apps.listingAll).toBe(false);
   checkbox.simulate("change");
   // The last call to fetchApps should list all the apps
-  const fetchCalls = defaultProps.fetchAppsWithUpdateInfo.mock.calls;
+  const fetchCalls = (defaultProps.fetchAppsWithUpdateInfo as jest.Mock).mock.calls;
```

Comment:
this kind of explicit changes is why I removed the type of the `defaultProps` object. I am not sure if it's worth to do so since actually the inner elements will differ from a `IAppListProps` for example.

Reply:
It's pretty easy to work around while still keeping things typed. I just updated with this change to just have a reference to the item first (and it's cleaner anyway):

```
 it("clicking 'List All' checkbox should trigger toggleListAll", () => {
+  const mockFetchAppsWithUpdateInfo = jest.fn();
+  const updatedProps: IAppListProps = {
+    ...defaultProps,
+    fetchAppsWithUpdateInfo: mockFetchAppsWithUpdateInfo,
+  }
   const apps = {
     isFetching: false,
     items: [],
@@ -220,12 +225,12 @@ it("clicking 'List All' checkbox should trigger toggleListAll", () => {
     ],
     listingAll: false,
   } as IAppState;
-  const wrapper = shallow(<AppList {...defaultProps} apps={apps} />);
+  const wrapper = shallow(<AppList {...updatedProps} apps={apps} />);
   const checkbox = wrapper.find('input[type="checkbox"]');
   expect(apps.listingAll).toBe(false);
   checkbox.simulate("change");
   // The last call to fetchApps should list all the apps
-  const fetchCalls = (defaultProps.fetchAppsWithUpdateInfo as jest.Mock).mock.calls;
+  const fetchCalls = mockFetchAppsWithUpdateInfo.mock.calls;
   expect(fetchCalls[fetchCalls.length - 1]).toEqual(["default", true]);
 });
```

Reply:
yes but you are still fooling typescript because `defaultProps.fetchAppWithUpdateInfo` will be reported as a `func()...` while it's a `Mock`.